reader_concurrency_semaphore.hh

```diff
@@ -317,7 +317,7 @@ public:
     /// Stop the reader_concurrency_semaphore and clear all inactive reads.
     ///
     /// Wait on all async background work to complete.
-    future<> stop() noexcept;
+    virtual future<> stop() noexcept;
```

Comment:
Do we expect derived classes to alter the content of `stop()` ever, or are they *all* expected to have a `stop()` like this:
```C++
virtual future<> stop() noexcept override {
    co_await reader_concurrency_semaphore::stop();
    // do additional stuff
    co_return;
}
```
If this is expected I'd suggest introducing a private virtual method, like `virtual stop_ext() {}` and calling it at the end of `reader_concurrency_semaphore::stop()`. Derived classes could override this to provide extra stop logic and we'd be guaranteed that no-one forgets to properly stop the common stuff.

Reply:
That sounds very reasonable to me, I will make the change.

Reply:
I have added two extension points, one before and one after the stop logic.